pkg/sql/tablewriter_update.go

```diff
@@ -55,8 +57,14 @@ func (tu *tableUpdater) rowForUpdate(
 	traceKV bool,
 ) (tree.Datums, error) {
 	tu.currentBatchSize++
+	var oth *row.OriginTimestampCPutHelper
```

Comment:
what if we instead had a little helper on `tableWriterBase` that fetches the `oth` variable? Then, we wouldn't need to add `evalctx` to tableUpdater etc.

I'm also wondering if we should pass around `oth` by value instead of by reference, to ensure that nobody accidentally adds an update or insert call that _should_ have a non empty oth var. It also means we have fewer callsites to audit e.g. (should r[owConverter ](https://github.com/jeffswenson/cockroach/blob/jeffswenson-ldr-fix-sql-take-2/pkg/sql/row/row_converter.go#L594)get `oth` passed?) If we pass `oth` by value, `oth.IsSet()` should still be correct and will only ever be true if the sql session is true.

Reply:
I moved initialization up to `pkg/sql/delete.go`, `pkg/sql/update.go`, `pkg/sql/insert.go`, and `pkg/sql/upsert.go`. One thing that was a little tricky is `startExec` isn't called uniformly for all run instances. So I generalized the init method to ensure the helper is always initialized if the session variable is set.